### PR TITLE
test/lib/user: add helper lib for doing things for and as a user

### PR DESCRIPTION
**Pull request by bboozzoo**
A common set of snippets that set up the user session and do things as a user.

### CODE REVIEW

File: tests/lib/desktop-portal.sh [hunk 1]
```diff
@@ -33,16 +33,12 @@ UseIn=spread
 EOF
 
     # Make sure the test user's XDG_RUNTIME_DIR exists
```
Comment (stolowski) [inline tests/lib/desktop-portal.sh]: This comment can be removed too.

File: tests/lib/user.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    if ! has_user_session_support ; then
+        # no session manager to kill
+        return 0
+    fi
+
+    systemctl stop "user@${TEST_UID}.service"
+}
+
+purge_user_session_data() {
+    if [ -n "$USER_RUNTIME_DIR" ] && [ -d "${USER_RUNTIME_DIR}" ]; then
```
Comment (stolowski) [inline tests/lib/user.sh]: Can the first arg be changed to "${USER_RUNTIME_DIR}" for consistency?

File: tests/lib/user.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+TEST_USER="test"
```
Comment (stolowski) [inline tests/lib/user.sh]: I was wondering if we have a central definition of test user anywhere (to replace this one), turns out we don't and simply use "test" anywhere :(

Comment (bboozzoo) [reply tests/lib/user.sh]: Part of the snapd-spread-tribal-knowledge.

**Comment by mvo5** [comment]
I guess the next step is to change a bunch of tests to actually use this stuff now :)

**Pull request merged**